src/test/java/com/amplitude/AmplitudeTest.java

```diff
@@ -382,22 +381,14 @@ public void testShouldWait()
     assertFalse(amplitude.shouldWait(event));
     amplitude.setRecordThrottledId(true);
     assertFalse(amplitude.shouldWait(event));
-    amplitude.setCallbacks(
-        new AmplitudeCallbacks() {
-          @Override
-          public void onLogEventServerResponse(Event event, int status, String message) {
-            callbackLatch.countDown();
-          }
-        });
     amplitude.logEvent(event);
     amplitude.logEvent(event2);
     amplitude.flushEvents();
     assertTrue(shouldWaitResultWithTimeout(amplitude, event, 1L, TimeUnit.SECONDS));
     assertFalse(amplitude.shouldWait(event2));
     barrier.await();
     assertTrue(latch.await(1L, TimeUnit.SECONDS));
-    assertTrue(callbackLatch.await(1L, TimeUnit.SECONDS));
-    assertFalse(shouldWaitResultWithTimeout(amplitude, event, 1L, TimeUnit.SECONDS));
+    assertTrue(shouldWaitResultWithTimeout(amplitude, event, 1L, TimeUnit.SECONDS));
```

Comment:
Maybe add a comment on why we flip on this?

Reply:
This is flipped because when it is a throttled rate limit response, it will retry until max retry times